Recipes by similar recipe

Marcobr93 opened the pull request:
Closes #15

Marcobr93 committed:
Lógica terminada

Juanlu1995 committed:
Merge branch 'master' into recipesBySimilarRecipe

Juanlu1995 committed:
cambios de carpetas

Marcobr93 committed:
Terminada la parte de comparar dos recetas

Marcobr93 committed:
Acabado

MeLlamoPablo commented on src/js/pages/search/getRecipesBySimilarRecipe.js:
```diff
@@ -0,0 +1,156 @@
+	return arraySimilarRecipes;
+}
+
+module.exports = getRecipesBySimilarRecipe.js;
```
Quita el .js de ahí

MeLlamoPablo commented on src/js/pages/search/getRecipesBySimilarRecipe.js:
```diff
@@ -0,0 +1,156 @@
+	 * los ingreditentes, sumaremos 1 a la variable numCoincidenciasIngreientes
+	 */
+	/* eslint-disable */
+	for (const ingredientes of receta.ingredient) {
```
Cuidado con los plurales y los singulares. Los nombres de los arrays siempre en plural, y los nombres de los valores únicos siempre en singular. Por lo tanto, sería `const ingredient of receta.ingredients`.

MeLlamoPablo commented on src/js/pages/search/getRecipesBySimilarRecipe.js:
```diff
@@ -0,0 +1,156 @@
+	 * de ambas recetas y los iremos comparando, si coinciden
+	 * los nombres, sumaremos 1 a la variable numCoincidenciasPalabras
+	 */
+	for (const palabras of receta.name) {
```
¡Cuidado! Los `for...of` se pueden aplicar a cualquier iterable. Los arrays son iterables. Las strings son iterables, pero... cuando iteras una string lo haces caracter por caracter, no palabra por palabra.

Para iterar palabra por palabra, hay que convertir `receta.name` en un array de `string`. Eso se puede hacer con `receta.name.split(" ")`.

MeLlamoPablo commented on src/js/pages/search/getRecipesBySimilarRecipe.js:
```diff
@@ -0,0 +1,156 @@
+	 * los nombres, sumaremos 1 a la variable numCoincidenciasPalabras
+	 */
+	for (const palabras of receta.name) {
+		for (const palabras2 of similarRecipe.name) {
```
Aquí pasa lo mismo

MeLlamoPablo commented on src/js/pages/search/getRecipesBySimilarRecipe.js:
```diff
@@ -0,0 +1,156 @@
+	 */
+	/* eslint-disable */
+	for (const ingredientes of receta.ingredient) {
+		for (const ingredientes2 of similarRecipe.ingredient) {
```
Aquí pasa lo mismo

MeLlamoPablo commented on src/js/pages/search/getRecipesBySimilarRecipe.js:
```diff
@@ -0,0 +1,156 @@
+	/**
+	 * Dependiendo del resultado obtenido, la puntuación será mayor o menor
+	 */
+	if (operacionIngredientes >= 50 && operacionIngredientes < 75) {
```
Hay una forma más sencilla y precisa de hacer esto: no multiplicar `numCoincidenciasIngredientes ` por 100, y sumar directamente ese número a `puntuacion`.

Lo que tienes ahora funciona (a falta de hacer los tests), así que si no quieres no tienes que cambiarlo, pero aún así yo lo veo más fácil.

MeLlamoPablo commented on src/js/pages/edit/addOrEditStepInRecipe.js:
```diff
@@ -0,0 +1,10 @@
+function addOrEditStepInRecipe(recipe, newStep, existingStepIndex) {
```
Parece que @Juanlu1995 se ha equivocado y ha puesto su función en tu rama, jajaja

En cualquier caso borradla alguno de los dos. Vamos a limitarnos a una función por rama/pull request.

MeLlamoPablo reviewed (changes requested):
Hay algunas cosas que fallan. Por el resto, a falta de subir los tests, lo veo bien.

Marcobr93 committed:
Corregido los fallos

MeLlamoPablo commented:
Closes #15

Marcobr93 committed:
Corregido los fallos2

MeLlamoPablo committed:
Eliminada la función addOrEditStepInRecipe de esta rama

MeLlamoPablo committed:
Arreglado para pasar ESLint

Por lo visto la guía que estamos usando de AirBNB no permite el uso
de for..of. Array.forEach es equivalente, así que podemos usar eso.

MeLlamoPablo committed:
Creado esqueleto del .spec.js
